### PR TITLE
Add descriptions to the levels in the career ladder.

### PR DESCRIPTION
**Pull request by mathias-abtion**
This pull request adds a few sentences describing what the expectations are for somebody at a given level in the career ladder.

The goal is to avoid that the career ladder simply becomes a checklist and to describe the characteristics we would like to see from somebody at a particular level.

### CODE REVIEW

File: career/ladder.md [hunk 1]
```diff
@@ -143,6 +165,12 @@ Here are some examples of other career ladders:
 - [ ] Can debug issues occurring across multiple levels of the stack.
 
 ### Level 4 - Senior Software Engineer
+As a Senior Software Engineer, you consistently show an ability to handle highly
+complex technical challenges across multiple axes without guidance. Your
+technical decisions are always guided by data and based on the highest value for
+all relevant parties. Product managers can rely on you when estimating the scale
+of a larger product, and your track record of turning big challenges into
+successes is apparent to the entire organization.
```
Comment (hcarreras) [inline career/ladder.md:173]: Very good!
We should also mention that as a senior software engineer, you've been around for a while. 
```suggestion
As a Senior Software Engineer, you consistently show an ability to handle highly
complex technical challenges across multiple axes without guidance. Your
technical decisions are always guided by data and based on the highest value for
all relevant parties. Your contributions are based on several years of experience where you had the chance to see the long-term effects on solutions you've designed and built. Product managers can rely on you when estimating the scale of a larger product, and your track record of turning big challenges into
successes is apparent to the entire organization.
```

File: career/ladder.md [hunk 2]
```diff
@@ -23,6 +23,13 @@ Here are some examples of other career ladders:
 ## Dev Values: Impactful, Simple, Test-Driven, Iterative, Communicative, Respectful, Skilled
 
 ### Level 1 - Associate Software Engineer
+As an Associate Software Engineer, you demonstrate an ability to complete work
+on a task-level in one or multiple apps, and not at least an eagerness to learn
```
Comment (hcarreras) [inline career/ladder.md:27]: May be a bit wordy.
```suggestion
on a task-level in one or multiple apps, and an eagerness to learn
```

File: career/ladder.md [hunk 3]
```diff
@@ -221,6 +255,12 @@ Here are some examples of other career ladders:
 - [ ] Anticipates technical issues at a product level and works with design to avoid them.
 
 ### Level 6 - Principal Software Engineer
+As a Principal Software Engineer, you play a critical role in how engineering at
+Abtion operates, and throughout your long career at Abtion, you have established
```
Comment (hcarreras) [inline career/ladder.md:259]: At this level, we also allow an area of ownership. E.g. tooling, security, design, process.
We need people at this level to be drivers. They are not only technical, they push Abtion forward.
I cannot find the right wording. We can pair on this one, as it is the toughest one.

```suggestion
Abtion operates. Throughout your long career at Abtion, you have established
```
A first step can be to give more importance to the fact that they'd be owning the way Abtion operates.

**Review by hcarreras — COMMENTED** [review]
Super good! I love these!

I would like to give a bit more thought to L6. Otherwise, it's good to go!

File: career/ladder.md (hunk 2 above)
Comment (MarkusAbtion) [inline career/ladder.md:27]: Lists with two items do not need a comma :) 
```suggestion
on a task-level in one or multiple apps and not at least an eagerness to learn
```

**Review by MarkusAbtion — APPROVED** [review]
My only suggestion was that one comma :shrug:
I think these descriptions are well-articulated and useful for guiding engineers up the ladder. 
Great work! ✨

File: career/ladder.md (hunk 2 above)
Comment (mathias-abtion) [reply career/ladder.md:27]: Thank you!

File: career/ladder.md (hunk 1 above)
Comment (mathias-abtion) [reply career/ladder.md:173]: Good point. I'll update it.

**Comment by mathias-abtion** [comment]
@hcarreras: I've rephrased level 6 and tried to take your feedback into account. What are your thoughts on the changes? Also, I am wondering whether it makes sense to have a separate document for [Roles](https://github.com/abtion/guidelines/blob/main/career/roles.md). Would it make sense to merge the two documents somehow?

**Review by hcarreras — APPROVED** [review]
Awesome! Let's get this merged
